Gems/Atom/Tools/MaterialCanvas/Code/Source/Window/MaterialCanvasGraphView.cpp

```diff
@@ -11,51 +11,14 @@
 
 namespace MaterialCanvas
 {
-    void MaterialCanvasGraphConstructPresets::InitializeConstructType(GraphCanvas::ConstructType constructType)
-    {
-        if (constructType == GraphCanvas::ConstructType::NodeGroup)
-        {
-            auto presetBucket = static_cast<GraphCanvas::NodeGroupPresetBucket*>(ModPresetBucket(GraphCanvas::ConstructType::NodeGroup));
-            if (presetBucket)
-            {
-                presetBucket->ClearPresets();
-
-                const AZStd::map<AZStd::string, AZ::Color> defaultPresetGroups = {
-                    { "Logic", AZ::Color(0.188f, 0.972f, 0.243f, 1.0f) },
-                    { "Function", AZ::Color(0.396f, 0.788f, 0.788f, 1.0f) },
-                    { "Output", AZ::Color(0.866f, 0.498f, 0.427f, 1.0f) },
-                    { "Input", AZ::Color(0.396f, 0.788f, 0.549f, 1.0f) } };
-
-                for (const auto& defaultPresetPair : defaultPresetGroups)
-                {
-                    if (auto preset = presetBucket->CreateNewPreset(defaultPresetPair.first))
-                    {
-                        const auto& presetSaveData = preset->GetPresetData();
-                        if (auto saveData = presetSaveData.FindSaveDataAs<GraphCanvas::CommentNodeTextSaveData>())
-                        {
-                            saveData->m_backgroundColor = defaultPresetPair.second;
-                        }
-                    }
-                }
-            }
-        }
-        else if (constructType == GraphCanvas::ConstructType::CommentNode)
-        {
-            auto presetBucket = static_cast<GraphCanvas::CommentPresetBucket*>(ModPresetBucket(GraphCanvas::ConstructType::CommentNode));
-            if (presetBucket)
-            {
-                presetBucket->ClearPresets();
-            }
-        }
-    }
-
     MaterialCanvasGraphView::MaterialCanvasGraphView(
-        const AZ::Crc32& toolId, const AZ::Uuid& documentId, const AtomToolsFramework::GraphViewConfig& graphViewConfig, QWidget* parent)
-        : AtomToolsFramework::GraphView(toolId, GraphCanvas::GraphId(), graphViewConfig, parent)
+        const AZ::Crc32& toolId,
+        const AZ::Uuid& documentId,
+        const AZStd::shared_ptr<AtomToolsFramework::GraphViewSettings>& graphViewSettingsPtr,
```

Comment:
If the intention is to pass `GraphViewSettings` without sharing or granting ownership, I suggest accepting a `const GraphViewSettings&` reference instead, or if this is a nullable parameter, just `const GraphViewSettings`. If however, this constructor intends to share ownership, passing `shared_ptr<GraphViewSettings>` by value is preferred, or if ownership is to be taken, an r-value `shared_ptr<GraphViewSettings>&&` instead. Finally, passing the shared pointer by reference really only makes sense if the function intends to manipulate the ownership of the caller (e.g. cause the caller to release its reference) or query the state of the shared pointer itself.

Reply:
Based on this feedback, passing by value sounds like the most appropriate option. Ownership is being shared to keep the instance alive and connected to the bus without having to use a weak pointer or check for validity. Most of the graph view settings members that were added should be accessible through the bus interface. Only the handful of members that were originally in the graph view config are excluded from that interface. I could make all of the members private and flesh out the interface for the additional members.